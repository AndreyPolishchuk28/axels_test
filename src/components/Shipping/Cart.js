import React, { useContext, useEffect } from 'react';
import { connect } from 'react-redux';

import { Container, QuantityWrapper, QuantityProducts } from '../../styled/Shipping/cart';
import { UserContext } from './UserContext';

const mapStateToProps = state => ({...state, product: state.products.product});

export const Cart = connect (mapStateToProps)(props => {
    const { quantity, setQuantity } = useContext(UserContext);

    useEffect(() => {
        if (props.product) {
            setQuantity(props.product.products.length)
        }
    });

    return(
        <Container>
            <svg width="1.5em" height="1.5em" viewBox="0 0 16 16" className="bi bi-cart" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
                 <path d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .491.592l-1.5 8A.5.5 0 0 1 13 12H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5zM3.102 4l1.313 7h8.17l1.313-7H3.102zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm-7 1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm7 0a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"/>
            </svg>
            <QuantityWrapper>
                <QuantityProducts>{quantity}</QuantityProducts>
            </QuantityWrapper>
        </Container>
    )
});
