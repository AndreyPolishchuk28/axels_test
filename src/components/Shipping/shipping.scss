.button-continue{
  background-color: #8752B2;
  border: 1px solid #8752B2;
}

.button-continue:hover{
  background-color: #8752B2;
}

.recipient{
  color: #8752B2;
}

.daytime{
  margin-top: 10px;
  margin-bottom: 20px;
}

.indent{
  margin-bottom: 10px;
}

.shipping-wrapper{
  padding-top: 20px;
}

.grey{
  background-color: #F1F3F6;
}

.img-size{
  height: 50px;
}

.wrapper-product{
  margin-top: 20px;
  padding-bottom: 10px;
  border-bottom: 1px solid #BDBFC1;
  width: 97%;
}

@media screen and (min-width: 1200px) {
  .wrapper-shipping-width{
    max-width: 450px;
  }

  .wrapper-order-width{
    max-width: 350px;
  }
}

@media screen and (min-width: 992px) {
  .wrapper-shipping-width{
    min-width: 450px;
    max-width: 450px;
  }

  .wrapper-order-width{
    min-width: 350px;
    max-width: 350px;
  }
}

.bg{
  background-color: #E6E9F0;
}

.shipping-color{
  color: #8752B2;
}

.lock{
  font-size: 20px;
}

.first-img{
  background-image: url("./img/backpack.jpg");
}

.second-img{
  background-image: url("./img/bag.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}

.third-img{
  background-image: url("./img/shoes.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}



