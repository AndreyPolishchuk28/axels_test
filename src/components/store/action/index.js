import {put, take, all} from 'redux-saga/effects'

const GET_PRODUCTS ='GET_PRODUCTS';
export const SET_PRODUCTS = 'SET_PRODUCTS';

export const getProducts = (payload) =>{
    return{
        type: GET_PRODUCTS,
        payload: payload
    }
};

export function* getProductsSaga() {
    while (true){
        try {
            yield take(GET_PRODUCTS);
            const req = yield fetch('https://demo3830727.mockable.io/products');
            const res = yield req.json();
            console.log(res);
            yield put({
                type:SET_PRODUCTS,
                payload: res
            })
        } catch (e) {
            console.log(e);
        }
    }
}


