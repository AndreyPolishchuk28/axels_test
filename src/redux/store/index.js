import {createStore, applyMiddleware, combineReducers} from 'redux';
import createSagaMiddleware from 'redux-saga';

import { rootSaga } from '../ducks/products';
import { productsReducer } from '../ducks/products';

const sagaMiddleware = createSagaMiddleware();

const saveToLocalStorage = state => {
    try {
        const serializedState = JSON.stringify(state);
        localStorage.setItem('state', serializedState);
    } catch (e) {
        console.log(e);
    }
};

const loadFromLocalStorage = () => {
    try {
        const serializedState = localStorage.getItem('state');
        if(serializedState === null) return undefined;
        return JSON.parse(serializedState);
    } catch (e) {
        console.log(e);
        return undefined;
    }
};
const persistedState = loadFromLocalStorage();

export const store = createStore(combineReducers({
    products: productsReducer
}), persistedState, applyMiddleware(sagaMiddleware));

store.subscribe(() => saveToLocalStorage(store.getState()));
sagaMiddleware.run(rootSaga);
