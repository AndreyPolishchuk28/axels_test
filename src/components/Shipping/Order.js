import React, { useEffect } from 'react';
import { Row, Col } from 'react-bootstrap';
import { connect } from 'react-redux';

import { Container, Title, Edit, FirstImg, NameProduct, Color, Quantity, Price, TotalPrice, TotalPricePurple, Conditions, UnderlineConditions } from '../../styled/Shipping/order';
import { getProducts } from '../../redux/ducks/products';

const mapStateToProps = state => ({...state});

export const Order = connect (mapStateToProps, { getProducts })(props => {
    const products = props.products.product;

    useEffect(() => {
        props.getProducts();
    },[]);

    const totalPrice = () => {
        let cost = 0;
        products.products.forEach(elem => cost += elem.price);
        return cost
    };

    return(
        <Container>
            <Row>
                <Col>
                    <Title>Order Summary</Title>
                </Col>
                <Col className='text-right'>
                    <Edit>edit order</Edit>
                </Col>
            </Row>
            {products.products && products.products.map(item =>
                <Row key={item.id} className='wrapper-product ml-1'>
                    <Col xl={2} md={2} sm={2} xs={2} className='p-0'>
                        <FirstImg src={item.img} />
                    </Col>
                    <Col xl={7} md={8} sm={6} xs={6} className='text-left img-size d-flex flex-column justify-content-between'>
                        <NameProduct>{item.name}</NameProduct>
                        <Color>{item.color}</Color>
                        <Quantity>Quantity: {item.quantity}</Quantity>
                    </Col>
                    <Col xl={3} md={2} sm={4} xs={4} className='text-right p-0'>
                        <Price>${item.price}</Price>
                    </Col>
                </Row>
            )
            }
            <Row className='wrapper-product ml-1'>
                <Col xl={8} md={8} sm={8} xs={8} className='d-flex flex-column justify-content-between text-left p-0'>
                    <TotalPrice>Subtotal</TotalPrice>
                    <TotalPrice>Shipping</TotalPrice>
                    <TotalPrice>Taxes</TotalPrice>
                </Col>
                <Col xl={4} md={4} sm={4} xs={4} className='text-right p-0 d-flex flex-column justify-content-between'>
                    {products.products && <TotalPrice>{totalPrice()}</TotalPrice>}
                    <TotalPrice>Free</TotalPrice>
                    {products.products && <TotalPrice>${products.taxes}</TotalPrice>}
                </Col>
            </Row>
            <Row className='mt-3'>
                <Col>
                    <TotalPricePurple>Total</TotalPricePurple>
                </Col>
                <Col className='text-right pb-5'>
                    {products.products && <TotalPricePurple>${totalPrice() + products.taxes}</TotalPricePurple>}
                </Col>
            </Row>
            <Row className='text-center mt-5'>
                <Col>
                    <Conditions>All purchases are subject to our <UnderlineConditions>Terms and Conditions</UnderlineConditions></Conditions>
                </Col>
            </Row>
        </Container>
    )
});
