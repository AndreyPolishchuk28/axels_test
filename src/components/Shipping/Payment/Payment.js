import React from "react";
import {Button, Col, Form, Row} from "react-bootstrap";
import {Arrow, Title} from "../ShippingInfo/ShippingInfo";
import {Order} from "../Order/Order";
import {Container} from "../ShippingInfo/ShippingInfo";
import {Link} from "react-router-dom";
import styled from "styled-components";

export const Payment = () =>{
    return(
        <Container className='container-fluid'>
            <Row className='d-flex justify-content-center bg'>
                <Col className='bg-white wrapper-shipping-width' md={6} sm={8} xs={10}>
                    <Row>
                        <Col className='d-flex align-items-center shipping-wrapper'>
                            <span>Shipping</span>
                            <Arrow/>
                            <span>Billing</span>
                            <Arrow/>
                            <span className='shipping-color'>Payment</span>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left'>
                            <Title>Payment</Title>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='d-flex '>
                            <i className="fas fa-lock lock"></i>
                            <SecureInfo className='ml-2'>This is a secure 128-bit SSL encrypted payment</SecureInfo>
                        </Col>
                    </Row>
                    <Form>
                        <Form.Group>
                            <Form.Label className='recipient'>Cardholder Name</Form.Label>
                            <Form.Control type="text" placeholder="Name as it appears on your card" />

                            <Form.Label className='recipient mt-4'>Card Number</Form.Label>
                            <Form.Control className='indent' type="text" placeholder="XXXX XXXX XXXX XXXX"/>

                            <Row>
                                <Col xl={3} lg={3} md={4} sm={4} xs={5}>
                                    <Form.Label className='recipient mt-3'>Expire Date</Form.Label>
                                    <Form.Control className='indent' type="text" placeholder="MM/YY"/>
                                </Col>
                                <Col xl={4} lg={4} md={4} sm={4} xs={6}>
                                    <Form.Label className='recipient mt-3'>Security Code</Form.Label>
                                    <Form.Control type="text" placeholder=""/>
                                </Col>
                            </Row>
                        </Form.Group>
                        <Row>
                            <Col md={7} sm={7} xs={7}>
                                <Link to='/print'>
                                    <Button className='button-continue mb-4' variant="primary" type="button" block>Pay Securely</Button>
                                </Link>
                            </Col>
                        </Row>
                    </Form>
                </Col>
                <Col className='grey wrapper-order-width' md={5} sm={8} xs={10}>
                    <Order/>
                </Col>
            </Row>
        </Container>
    )
};

const SecureInfo = styled.p`
    color: #CDCDCD;
    font-size: 18px;
`;
