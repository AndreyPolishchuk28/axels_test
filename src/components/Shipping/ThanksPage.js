import React, { useEffect, useState } from 'react';
import { Col, Row } from 'react-bootstrap';
import { connect } from 'react-redux';

import { Order } from './Order';
import { Container } from '../../styled/similarStyle';
import { Title, OrderNumber, EmailInfo, Email, Date, Print } from '../../styled/Shipping/thanksPage';
import { shippingInfo } from '../../redux/ducks/products';

const mapStateToProps = state => ({...state});

export const ThanksPage = connect(mapStateToProps, { shippingInfo })(props => {
    const [email, setEmail] = useState();

    const returnLastEmail = (arr) => arr[arr.length - 2];

    useEffect(() => {
        if (props.products){
            setEmail(returnLastEmail(props.products.userAddress))
        }
    },[props.products]);

    return (
        <Container className='container-fluid'>
            <Row className='d-flex justify-content-center bg'>
                <Col className='bg-white wrapper-shipping-width' md={6} sm={8} xs={10}>
                    <Row>
                        <Col className='text-left mt-5 ml-3'>
                            <Title>Thank you for your order!</Title>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left ml-3'>
                            <OrderNumber>Order number is: 188787788</OrderNumber>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left ml-3 mr-5'>
                            {email && <EmailInfo>Your will receive an email confirmation shortly to <Email>{email.email}</Email></EmailInfo>}
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left ml-3'>
                            <EmailInfo className='mb-0'>Estimated delivery Day is</EmailInfo>
                            <Date>Friday 1st April 2016</Date>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left ml-3 mt-2'>
                            <Print>Print Recipe</Print>
                        </Col>
                    </Row>
                </Col>
                <Col className='grey wrapper-order-width' md={5} sm={8} xs={10}>
                    <Order/>
                </Col>
            </Row>
        </Container>
    );
});
