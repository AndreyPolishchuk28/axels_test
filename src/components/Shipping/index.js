import React from "react";
import {Arrow, ShippingInfo, Title} from "./ShippingInfo/ShippingInfo";
import {Row, Col, Button, Form} from 'react-bootstrap';
import {Order} from "./Order/Order";
import styled from "styled-components";
import {Link} from "react-router-dom";


export const Shipping = () =>{
    return(
        <Container className='container-fluid'>
            <Row className='d-flex justify-content-center bg'>
                <Col className='bg-white wrapper-shipping-width' md={6} sm={8} xs={10}>
                    <Row>
                        <Col className='d-flex align-items-center shipping-wrapper'>
                            <span className='shipping-color'>Shipping</span>
                            <Arrow/>
                            <span>Billing</span>
                            <Arrow/>
                            <span>Payment</span>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left'>
                            <Title>Shipping Info</Title>
                        </Col>
                    </Row>
                    <Form>
                        <Form.Group>
                            <Form.Label className='recipient mb-3'>Recipient</Form.Label>
                            <Form.Control type="text" placeholder="Full Name" />
                            <Row className='daytime'>
                                <Col md={8} sm={8} xs={8}>
                                    <Form.Control type="text" placeholder="Daytime Phone" />
                                </Col>
                                <Col>
                                    <Form.Text className="text-muted">
                                        For delivery quetions only
                                    </Form.Text>
                                </Col>
                            </Row>
                            <Form.Label className='recipient mb-0'>Address</Form.Label>
                        </Form.Group>
                    </Form>

                    <ShippingInfo/>
                    <Row>
                        <Col md={7} sm={7} xs={7}>
                            <Link to='/billing'>
                                <Button className='button-continue mb-4' variant="primary" type="button" block>Continue</Button>
                            </Link>
                        </Col>
                    </Row>
                </Col>
                <Col className='grey wrapper-order-width' md={5} sm={8} xs={10}>
                    <Order/>
                </Col>
            </Row>
        </Container>
    )
};

export const Container = styled.div`
    background-color: #E6E9F0;
    padding-top: 15px;
    height: 100vh;
`;
