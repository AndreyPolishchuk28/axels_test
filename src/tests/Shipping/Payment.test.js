import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import { Payment } from '../../components/Shipping/Payment';
import { store } from '../../redux/store';
import { act } from 'react-dom/test-utils';

describe('Payment', () => {
    it('should render Payment component', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <Payment />
                </Router>
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('button continue payment', () => {
        const component = mount(
            <Provider store={store}>
                <Router>
                    <Payment />
                </Router>
            </Provider>
        );

        const handleBtn = component.find('button').props();
        const somethingSpy = jest.spyOn(handleBtn, 'onClick');
        act(() => handleBtn.onClick());
        expect(somethingSpy).toHaveBeenCalledTimes(1);
    });

    it('correct inputs data', () => {
        const component = mount(
            <Provider store={store}>
                <Router>
                    <Payment />
                </Router>
            </Provider>
        );

        let cardHolderName = component.find('input').at(0);
        let cardNumber = component.find('input').at(1);
        let expireDate = component.find('input').at(2);
        let securityCode = component.find('input').at(3);

        cardHolderName.simulate('change', {target: {name: 'cardHolderName', value: 'test'} });
        cardNumber.simulate('change', {target: {name: 'cardNumber', value: 'test'} });
        expireDate.simulate('change', {target: {name: 'expireDate', value: 'test'} });
        securityCode.simulate('change', {target: {name: 'securityCode', value: 'test'} });

        cardHolderName = component.find('input').at(0);
        cardNumber = component.find('input').at(1);
        expireDate = component.find('input').at(2);
        securityCode = component.find('input').at(3);

        expect(cardHolderName.props().value).toBe('test');
        expect(cardNumber.props().value).toBe('test');
        expect(expireDate.props().value).toBe('test');
        expect(securityCode.props().value).toBe('test');
    });
});
