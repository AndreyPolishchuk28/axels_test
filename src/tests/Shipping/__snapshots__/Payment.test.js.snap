// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should render Payment component 1`] = `
<div
  className="sc-AxjAm emnRQd container-fluid"
>
  <div
    className="d-flex justify-content-center bg row"
  >
    <div
      className="bg-white wrapper-shipping-width col-md-6 col-sm-8 col-10"
    >
      <div
        className="row"
      >
        <div
          className="d-flex align-items-center shipping-wrapper col"
        >
          <span>
            Shipping
          </span>
          <div
            className="sc-AxirZ gwOTvu"
          />
          <span>
            Billing
          </span>
          <div
            className="sc-AxirZ gwOTvu"
          />
          <span
            className="shipping-color"
          >
            Payment
          </span>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="text-left col"
        >
          <h2
            className="sc-AxiKw iYcDPJ"
          >
            Payment
          </h2>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="d-flex  col"
        >
          <i
            className="fas fa-lock lock"
          />
          <p
            className="sc-AxhCb fHTAEz ml-2"
          >
            This is a secure 128-bit SSL encrypted payment
          </p>
        </div>
      </div>
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="recipient form-label"
          >
            Cardholder Name
          </label>
          <input
            className="form-control"
            name="cardHolderName"
            onChange={[Function]}
            placeholder="Name as it appears on your card"
            type="text"
            value=""
          />
          <small
            className="errors form-text"
          />
          <label
            className="recipient mt-4 form-label"
          >
            Card Number
          </label>
          <input
            className="indent form-control"
            name="cardNumber"
            onChange={[Function]}
            placeholder="XXXX XXXX XXXX XXXX"
            type="text"
            value=""
          />
          <small
            className="errors form-text"
          />
          <div
            className="row"
          >
            <div
              className="col-xl-3 col-lg-3 col-md-4 col-sm-4 col-5"
            >
              <label
                className="recipient mt-3 form-label"
              >
                Expire Date
              </label>
              <input
                className="indent form-control"
                name="expireDate"
                onChange={[Function]}
                placeholder="MM/YY"
                type="text"
                value=""
              />
              <small
                className="errors form-text"
              />
            </div>
            <div
              className="col-xl-4 col-lg-4 col-md-4 col-sm-4 col-6"
            >
              <label
                className="recipient mt-3 form-label"
              >
                Security Code
              </label>
              <input
                className="form-control"
                name="securityCode"
                onChange={[Function]}
                placeholder=""
                type="text"
                value=""
              />
              <small
                className="errors form-text"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-7 col-sm-7 col-7"
          >
            <button
              className="button-continue mb-4 btn btn-primary btn-block"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Pay Securely
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      className="grey wrapper-order-width col-md-5 col-sm-8 col-10"
    >
      <div
        className="sc-AxhUy jgEmor"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <h3
              className="sc-AxgMl jDLWTK"
            >
              Order Summary
            </h3>
          </div>
          <div
            className="text-right col"
          >
            <span
              className="sc-AxheI cCnoCk"
            >
              edit order
            </span>
          </div>
        </div>
        <div
          className="wrapper-product ml-1 row"
        >
          <div
            className="d-flex flex-column justify-content-between text-left p-0 col-xl-8 col-md-8 col-sm-8 col-8"
          >
            <span
              className="sc-fzplWN ZTybQ"
            >
              Subtotal
            </span>
            <span
              className="sc-fzplWN ZTybQ"
            >
              Shipping
            </span>
            <span
              className="sc-fzplWN ZTybQ"
            >
              Taxes
            </span>
          </div>
          <div
            className="text-right p-0 d-flex flex-column justify-content-between col-xl-4 col-md-4 col-sm-4 col-4"
          >
            <span
              className="sc-fzplWN ZTybQ"
            >
              Free
            </span>
          </div>
        </div>
        <div
          className="mt-3 row"
        >
          <div
            className="col"
          >
            <span
              className="sc-fznyAO gQvMrU"
            >
              Total
            </span>
          </div>
          <div
            className="text-right pb-5 col"
          />
        </div>
        <div
          className="text-center mt-5 row"
        >
          <div
            className="col"
          >
            <p
              className="sc-fznKkj jgmjte"
            >
              All purchases are subject to our 
              <span
                className="sc-fznZeY dpajSd"
              >
                Terms and Conditions
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
